File: ecmascript/bob/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

class Bob {
  hey(message) {
   if (this.isYelling(message)) return "Whoa, chill out!";
   else if (this.isQuestion(message)) return "Sure.";
   else if (this.isSilence(message)) return "Fine. Be that way!";
   return "Whatever.";
  }

  isYelling = (input) => { return input === input.toUpperCase() && input.match(/[a-z]/i) };

  isQuestion = (input) => { return input.endsWith("?"); };

  isSilence = (input) => { return input.trim().length === 0; };
}


export default Bob;

File: elixir/bob/bob.exs
defmodule Bob do
def hey(input) do
    cond do
        question?(input) -> "Sure."
        empty?(input) -> "Fine. Be that way!"
        shouting?(input) -> "Whoa, chill out!"
        true -> "Whatever."
    end
  end

  def empty?(input), do: String.strip(input) == ""

  def shouting?(input), do: String.upcase(input) == input && valid_speech(input)

  def question?(input), do: String.last(input) == "?"

  def valid_speech(input), do: input =~ ~r/[[:alpha:]]/
end
